src/protocol/primitives/uint16.ts

```diff
@@ -0,0 +1,18 @@
+import type { ReadBuffer } from '../serialization/read-buffer';
+import { Deserializable, type Serializable } from '../serialization/serialization';
+import type { WriteBuffer } from '../serialization/write-buffer';
+
+export class UInt16 extends Deserializable implements Serializable {
```

Comment:
The type of the UInt16 is not specified in the Kafka Protocol Guide. Only specified integer types are `INT8, INT16, INT32, INT64, UINT32, VARINT and VARLONG`